### PR TITLE
feat: Add CouchDB implementation for block storage

### LINKED ISSUE #32: Add CouchDB implementation for block storage

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

closes #32

### CODE REVIEW

File: mod/peer/blkstorage/store_test.go [hunk 1]
```diff
@@ -7,26 +7,16 @@ SPDX-License-Identifier: Apache-2.0
...
 	"github.com/hyperledger/fabric/core/ledger/ledgerconfig"
-	"github.com/spf13/viper"
 	"github.com/stretchr/testify/require"
+	"github.com/trustbloc/fabric-peer-ext/mod/peer/testutil"
 )
 
 func TestNewProvider(t *testing.T) {
-	cleanup := setupPath(t)
-	defer cleanup()
+	_, _, destory := testutil.SetupExtTestEnv()
```
Comment (bstasyszyn) [inline mod/peer/blkstorage/store_test.go]: destroy

File: pkg/blkstorage/cdbblkstorage/cdb_checkpoint.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cdbblkstorage
+
+import (
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
```
Comment (bstasyszyn) [inline pkg/blkstorage/cdbblkstorage/cdb_checkpoint.go]: Should we be using gogo/protobuf?

File: mod/peer/blkstorage/store_test.go (hunk 1 above)
Comment (fqutishat) [reply mod/peer/blkstorage/store_test.go]: done

File: pkg/blkstorage/cdbblkstorage/cdb_checkpoint.go (hunk 2 above)
Comment (fqutishat) [reply pkg/blkstorage/cdbblkstorage/cdb_checkpoint.go]: I remove it

**Comment by fqutishat** [comment]
verify-unitest

**Pull request merged**